Client/Coordinators/Library/BookmarksCoordinator.swift

```diff
@@ -0,0 +1,68 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Storage
+
+protocol BookmarksCoordinatorDelegate: AnyObject {
+    func start(from cell: BookmarksFolderCell)
+
+    func showBookmarkDetail(for node: FxBookmarkNode, folder: FxBookmarkNode)
+}
+
+class BookmarksCoordinator: BaseCoordinator, BookmarksCoordinatorDelegate {
+    // MARK: - Properties
+
+    private let profile: Profile
+    private weak var parentCoordinator: LibraryCoordinatorDelegate?
+
+    // MARK: - Initializers
+
+    init(
+        router: Router,
+        profile: Profile,
+        parentCoordinator: LibraryCoordinatorDelegate?
+    ) {
+        self.profile = profile
+        self.parentCoordinator = parentCoordinator
+        super.init(router: router)
+    }
+
+    // MARK: - BookmarksCoordinatorDelegate
+
+    func start(from cell: BookmarksFolderCell) {
+        if let itemData = cell as? BookmarkItemData {
+            start(itemData: itemData)
+        } else if let folderData = cell as? FxBookmarkNode {
+            start(folderData: folderData)
+        }
+    }
+
+    private func start(folderData: FxBookmarkNode) {
+        let viewModel = BookmarksPanelViewModel(profile: profile, bookmarkFolderGUID: folderData.guid)
+        let controller = BookmarksPanel(viewModel: viewModel)
+        controller.bookmarkCoordinatorDelegate = self
+        if folderData.isRoot, let localizedString = LocalizedRootBookmarkFolderStrings[folderData.guid] {
+            controller.title = localizedString
+        } else {
+            controller.title = folderData.title
+        }
+        router.push(controller)
+    }
+
+    private func start(itemData: BookmarkItemData) {
+        guard let url = URIFixup.getURL(itemData.url) ?? profile.searchEngines.defaultEngine?.searchURLForQuery(itemData.url) else { return }
```

Comment:
I am not sure I understand this case. We want to start `BookmarkItemData`, which means we want to open a bookmark? This doesn't push any views, which feels like this should be in the VC rather than the coordinator. Coordinators are meant to only handle navigation flow.

Reply:
Ok then i should use this function just for bookmark folder cell